parser.py

```diff
@@ -1 +1,40 @@
-# TODO Xin - Add methods here that we will need. Try to think carefully about organization and how to separate functionality effectively.
\ No newline at end of file
+# TODO Xin - Add methods here that we will need. Try to think carefully about organization and how to separate functionality effectively.
+from bs4 import BeautifulSoup as bs
+from bs4 import SoupStrainer as ss
+import re
+import urllib2 as ul
+ 	
+def links(webpage):
```

Comment:
Let's rename this to something more descriptive. Like "get_500_companies" or "get_names_and_links"